studiopress/genesis-custom-blocks
# Template editor React implementation

Updated April 22nd

#### Changes

* This syntax in the template: `{{cta-text}}` would be: `block_field( 'cta-text' )`
* But that's the only dynamic syntax. This doesn't have conditionals or for loops, it's not even Mustache
* There's a Markup and CSS editor:
<img width="779" alt="markup-button-here" src="https://user-images.githubusercontent.com/4063887/115792722-63ce1380-a390-11eb-81d0-1d3fe91fc417.png">

#### Testing Instructions

1. `composer install && npm install && npm run dev`
2. `/wp-admin` > Custom Blocks > Add New
3. Give the block any title you want: 
![your-blocks-](https://user-images.githubusercontent.com/4063887/115813490-dd79f780-a3b8-11eb-971d-71f151638f77.png)
4. Add a new field, maybe named "Your Example Field": 
![your-example-field](https://user-images.githubusercontent.com/4063887/115813581-08fce200-a3b9-11eb-8ce2-d2f41f5521f5.png)
5. Click the 'Template Editor' button
6. Expected: It should reference the field you created: 
![your-example-one](https://user-images.githubusercontent.com/4063887/115813693-3fd2f800-a3b9-11eb-8d28-2dc46e2eac8f.png)
7. Enter any markup you want, maybe like:

```html
<div class="foo-class">
    <p class="baz">
        {{your-example-field}}
    </p>
</div>
```
![some-markup](https://user-images.githubusercontent.com/4063887/115813858-9c361780-a3b9-11eb-8a6a-d0c7155fe516.png)

8. Click 'Save draft' and refresh the page
9. Expected: The markup should still be there
10. Click the 'CSS' button and enter any CSS
11. Click 'Save draft'
12. Expected: it's still there

Out of scope: 'Front-end Preview' will not work for this. That's in an upcoming PR.  This doesn't do anything with the markup or CSS editor yet, it just saves it.

Fixes [GF-2974](https://wpengine.atlassian.net/browse/GF-2974)
